spec/features/applications/show_spec.rb

```diff
@@ -88,6 +101,61 @@
 
         expect(page).to have_content(@nigel.status)
       end
+
+      it 'Then I see a section on page to add a pet to this application' do
+        visit "/applications/#{@nigel.id}"
+
+        expect(page).to have_content('Add a pet to this application')
+        expect(page).to have_field('Search for Pets')
+      end
+
+      it 'Under the search bar I see any Pet whose name matches my search' do
+        visit "/applications/#{@nigel.id}"
+
+        fill_in('Search for Pets', with: "#{@pet_1.name}")
+        click_button('Submit')
+
+        within('section#add_pet') do
+          expect(page).to have_content("#{@pet_1.name}")
+        end
+      end
+
+      it 'Can display other pets' do
+        visit "/applications/#{@nigel.id}"
+
+        fill_in('Search for Pets', with: "#{@pet_2.name}")
+        click_button('Submit')
+
+        within('section#add_pet') do
+          expect(page).to have_content("#{@pet_2.name}")
+        end
+      end
+
+      it 'Tests for partial matches' do
+        visit "/applications/#{@nigel.id}"
+
+        fill_in('Search for Pets', with: "a")
+        click_button('Submit')
+
+        within('section#add_pet') do
+          expect(page).to have_content("#{@pet_1.name}")
+          expect(page).to have_content("#{@pet_2.name}")
+          expect(page).to have_content("#{@pet_3.name}")
+        end
+      end
+
+      it 'Shows nothing if name != any pets' do
```

Comment:
I believe this covers most edge cases I can think of including sad path.  Let's continue user stories and play with the rails server local host side to see if we can come up with any more bugs.